codice/ddf
# DDF-4237 Extracts JAXB generation to another repository

#### What does this PR do?
Extracts generated JAXB resources to another repository and brings that in as a dependency.

##### Special Note
*This is dependent upon codice/ddf-jaxb#2 and cannot be merged until that change has been merged and the DDF-JAXB project has been deployed. Then, the version referenced in the reactor pom of this repository - 1.0.0-SNAPSHOT - will need to be updated to the release version.*

#### Who is reviewing it? 

@brjeter @paouelle @blen-desta 

#### Select relevant component teams: 
@codice/io 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.

@clockard
@stustison

#### How should this be tested?


1. Full CI build must pass
2. Install DDF and test affected functionality
    * Create CSW sources and ensure they work correctly
    * Create WFS 1.0, 1.1, and 2.0 sources and ensure they work correctly
    * Ensure security - CAS and XACML in particular - still work
    * Ensure that metacard transformations (ingests, etc.) still work correctly

#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-4237](https://codice.atlassian.net/browse/DDF-4237)

#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
